src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,67 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health;
+  name;
+
+  constructor(name = '', health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (this.health > 0) {
+      Animal.alive.push(this);
+    }
+  }
+
+  takeDamage(damage) {
+    this.health -= damage;
+
+    if (this.health <= 0) {
+      const index = Animal.alive.findIndex((animal) => animal === this);
+
+      Animal.alive.splice(index, 1);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden;
+
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  takeDamage(damage) {
+    if (!this.hidden) {
+      super.takeDamage(damage);
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  biteStrength;
+
+  constructor(name, health, biteStrength = 50) {
+    super(name, health);
+
+    this.biteStrength = biteStrength;
+  }
+
+  bite(target) {
+    if (!(target instanceof Herbivore)) {
+      return;
+    }
+
+    target.takeDamage(this.biteStrength);
```

Comment:
The bite method should check if the target is not an instance of Carnivore and also if the target is not hidden before applying damage. Currently, it only checks if the target is not an instance of Herbivore, which is not sufficient according to the task requirements.

Reply:
I **DO** check if the target is hidden. See `Herbivore.takeDamage()` method.